airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -89,7 +89,7 @@ class ReportStream(BasicAmazonAdsStream, ABC):
     Common base class for report streams
     """
 
-    primary_key = ["profileId", "recordType", "reportDate", "updatedAt"]
+    primary_key = ["profileId", "recordType", "reportDate", "updatedAt", "recordId"]
```

Comment:
Are all of these still needed for uniqueness? Trying to understand where `reportDate` fits in here. I could see "profileid's report about recordtype:recordID" being an identifier for one report, and reportDate potentially being a way to distinguish multiple reports about the same recordtype:recordID, but couldn't you create multiple on the same date?

And doesn't `updatedAt` as part of the PK make updating an existing report effectively turn it into a new report from the perspective of the deduper? (I think this was mentioned in the issue)

Reply:
You are right. We need `updatedAt` to distinguish multiple syncs of the same report done at the same date. But `updatedAt` is a `reportDate` + `datetime`, so we have `reportDate` added two times to primary key for no reason. It breaks nothing for now but should be avoided

Reply:
> distinguish multiple syncs of the same report done at the same date

In that case its still the same report though right? In which case it should probably be a cursor, not part of the primary key? Unless I'm missing something

Reply:
Yes, this is true. https://github.com/airbytehq/airbyte/pull/21677/files#diff-4e3b79abb6fc723d5f9e51b1ff286901bb4bfb259b525fdcd02b5e1a44ed5d10R100

Reply:
This doesn't seem to link to a diff anymore - I'm suggesting that I think `updatedAt` doesn't belong in the composite primary key

Reply:
I agree with Ella. `updatedAt` should be a cursor field used to track updates on the daily report (keyed off the `reportDate` field)

Reply:
Please keep in mind that `updatedAt` is artificial field which is added when real sync was done 
it's possible to have such object:

```json
{
 "reportDate": "2023-01-10"
 "updatedAt": "2023-01-26T00:00:00Z"
}
```

it means we synced report object for `2023-01-10`, but real sync was done `2023-01-26T00:00:00Z`

We use
```
cursor_field: reportDate
```

to keep track what last report was synced.

Reply:
So.. let me start with one implicit problem we're trying to solve as well. Since the cursor field we have is of a `date` type and the dedup sync may be running once in a couple of hours, we're not sure which record with the same primary key and cursor value would be deleted and which would be stored. If the one with bigger `emitted_at` value is stored, as Edward [supposes](https://airbytehq.slack.com/archives/C02UF50V9HA/p1674834152323399?thread_ts=1674819243.778729&cid=C02UF50V9HA), then we're fine to drop `updatedAt` both from the primary key and from the record. Otherwise we can not guarantee that the latest data is stored, thus can not guarantee correctness of a dedup sync mode and need to remove both `updatedAt`, `recordId` and the primary key itself.

@erohmensing @girarda please let me know your opinion

Reply:
btw, `updatedAt` is an artificial field and is actually a time when the record is emitted

Reply:
> If the one with bigger emitted_at value is stored, as Edward [supposes](https://airbytehq.slack.com/archives/C02UF50V9HA/p1674834152323399?thread_ts=1674819243.778729&cid=C02UF50V9HA), then we're fine to drop updatedAt both from the primary key and from the record.

This sounds like the expected behavior to me, and I think we'd be seeing a lot more on call issues if it weren't the case. Can we test that that is what happens when the cursor field is date type and being synced on the level of hours? If so, I think dropping `updatedAt` from the PK, and from the record if emitted_at is already inserted into the record

Reply:
makes sense, I will test it

Reply:
So I checked it and yes, it works as expected so I'm gonna remove `updatedAt` field

Reply:
done